libs/api/libbpf_map.cpp

```diff
@@ -354,9 +358,62 @@ bpf_map_get_next_id(uint32_t start_id, uint32_t* next_id)
 
 typedef struct ring_buffer
 {
-    std::vector<ring_buffer_subscription_t*> subscriptions;
+    ring_buffer() : ctx(nullptr), sample_cb(nullptr), active_subscription_count(0), free_inprogress(false) {}
+    ~ring_buffer();
+
+    std::vector<ebpf_map_subscription_t*> subscriptions;
+    void* ctx;
```

Comment:
why are these new fields necessary? In ring buffer the callback function and the context were stored in the subscription object itself. Adding these fields in the ring_buffer and perf_buffer structs is introducing many complexities, including a "wrapper callback"; and making the unsubscribe call blocking by introducing a rundown. The original unsubscribe API is unblocking and it makes the programming model simpler for apps.